compiler/modules/nback/init.bal

```diff
@@ -279,28 +286,37 @@ function addRecordSubtypeDefn(InitModuleContext cx, string symbol, string[] fiel
     return cx.llMod.addGlobal(llType, symbol, initializer=initValue, align=SUBTYPE_ALIGN, isConstant=true, linkage="internal");
 }
 
-function getListSetFunc(InitModuleContext cx) returns llvm:FunctionDecl {
-    llvm:FunctionDecl? existing = cx.listSetFunc;
-    if existing == () {
-        llvm:FunctionDecl decl = cx.llMod.addFunctionDecl(mangleRuntimeSymbol("list_set"), llListSetFuncType);
-        cx.listSetFunc = decl;
-        return decl;
+function getListAccessor(InitModuleContext cx, ListRepr? repr) returns ListAccessors {
+    if repr == () {
+        ListAccessors? existing = cx.genericListAccessors;
+        if existing == () {
+            ListAccessors a = createListAccessor(cx, LIST_GENERIC_REPR);
+            cx.genericListAccessors = a;
+            return a;
+        }
+        else {
+            return existing;
+        }
     }
     else {
-        return existing;
+        return createListAccessor(cx, repr);
     }
 }
 
-function getListGetFunc(InitModuleContext cx) returns llvm:FunctionDecl {
-    llvm:FunctionDecl? existing = cx.listGetFunc;
-    if existing == () {
-        llvm:FunctionDecl decl = cx.llMod.addFunctionDecl(mangleRuntimeSymbol("list_get"), llListGetFuncType);
-        cx.listGetFunc = decl;
-        return decl;
-    }
-    else {
-        return existing;
+final readonly & string[] LIST_ACCESSOR_RETURN_TYPES = ["tagged", "int", "float"];
```

Comment:
This sort of thing is fine in all caps, because it's only a JBUG that is stopping it being a const.